src/lib.rs

```diff
@@ -259,10 +261,10 @@ pub trait ImguiRenderLoop {
     /// Called during the window procedure.
     fn on_wnd_proc(&self, _hwnd: HWND, _umsg: u32, _wparam: WPARAM, _lparam: LPARAM) {}
 
-    /// If this method returns `true`, the WndProc function will not call the
-    /// procedure of the parent window.
-    fn should_block_messages(&self, _io: &Io) -> bool {
-        false
+    /// Sets the types of window message that
+    /// you do not want to propagate to the main window
+    fn set_message_filter(&self, _io: &Io) -> MessageFilter {
```

Comment:
Could we change the name here? `set_` prefix usually implies some sort of internal mutation, whereas this method returns a value. Just `message_filter` could work, but I'm willing to consider other ideas.

What's important is that the user sees the method name and immediately thinks "ok, here I need to implement the logic that will tell the system which messages to ignore according to the current state".